src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -473,7 +473,7 @@ public BoundReturnStatement Return(BoundExpression expression = null)
                 }
             }
 
-            return new BoundReturnStatement(Syntax, CurrentFunction.RefKind, expression) { WasCompilerGenerated = true };
+            return new BoundReturnStatement(Syntax, BoundReturnStatement.StatementRefKind(CurrentFunction.RefKind, expression?.GetRefKind() ?? RefKind.None), expression) { WasCompilerGenerated = true };
```

Comment:
>BoundReturnStatement.StatementRefKind(CurrentFunction.RefKind, expression?.GetRefKind() ?? RefKind.None) [](start = 52, length = 104)

I don't think this should change. I would be surprised to see Ref expression here.